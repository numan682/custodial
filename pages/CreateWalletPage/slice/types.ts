/* --- STATE --- */
export interface CreateWalletState {
  mnemonic: string[]
  checkbox: boolean
}
