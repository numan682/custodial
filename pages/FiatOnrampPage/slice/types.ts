export interface FiatOnrampState {
  thirdPartyAcknowledged: boolean
}
