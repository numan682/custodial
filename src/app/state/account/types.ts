import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  debonding: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  delegations: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
