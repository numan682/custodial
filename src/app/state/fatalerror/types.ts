/* --- STATE --- */
export interface FatalErrorPayload {
  message: string
  stack?: string
  sagaStack?: string
}

export interface FatalErrorState {
  error?: FatalErrorPayload
}
