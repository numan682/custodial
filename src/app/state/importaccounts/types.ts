import { ErrorPayload } from 'types/errors'
import { WalletType } from '../wallet/types'
import { BalanceDetails } from '../account/types'

/* --- STATE --- */
export interface ImportAccountsListAccount {
  address: string
  balance?: BalanceDetails
  path: number[]
  pathDisplay: string
  privateKey?: string
  publicKey: string
  selected: boolean
  type: WalletType
}

export enum ImportAccountsStep {
  Idle = 'idle',
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
}

export interface ImportAccountsState {
  accounts: ImportAccountsListAccount[]
  error?: ErrorPayload
  showAccountsSelectionModal: boolean
  accountsSelectionPageNumber: number
  step: ImportAccountsStep
}
