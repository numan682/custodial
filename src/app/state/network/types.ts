/* --- STATE --- */

export type NetworkType = 'local' | 'testnet' | 'mainnet'

export interface NetworkState {
  selectedNetwork: NetworkType

  /** Ticker symbol (ROSE, TEST, etc.) */
  ticker: string

  /** chainContext / Genesis Hash */
  chainContext: string

  /** Current epoch */
  epoch: number

  /** Minimum staking amount */
  minimumStakingAmount: number
}
