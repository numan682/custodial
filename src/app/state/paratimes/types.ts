import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload } from 'types/errors'
import { ParaTime } from '../../../config'

export enum TransactionFormSteps {
  TransferType = 'transferType',
  ParaTimeSelection = 'paraTimeSelection',
  TransactionRecipient = 'transactionRecipient',
  TransactionAmount = 'transactionAmount',
  TransactionConfirmation = 'transactionConfirmation',
  TransactionSummary = 'transactionSummary',
  TransactionError = 'transactionError',
}

export enum TransactionTypes {
  Deposit = 'deposit',
  Withdraw = 'withdraw',
}

export interface TransactionForm {
  amount: string
  confirmTransfer: boolean
  confirmTransferToValidator: boolean
  confirmTransferToForeignAccount: boolean
  defaultFeeAmount: string
  ethPrivateKey: string
  feeAmount: string
  feeGas: string
  paraTime?: ParaTime
  recipient: string
  type: TransactionTypes | undefined
}

export interface ParaTimesState {
  balance: StringifiedBigInt
  isLoading: boolean
  transactionError?: ErrorPayload
  transactionForm: TransactionForm
  transactionFormStep: TransactionFormSteps
}

export type Runtime = {
  address: string
  id: string
  decimals: number
  feeGas: bigint
}

export type ParaTimeTransaction = Pick<
  TransactionForm,
  'amount' | 'ethPrivateKey' | 'feeAmount' | 'feeGas' | 'recipient' | 'type'
>
