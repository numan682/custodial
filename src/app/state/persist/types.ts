import { RootState } from 'types'
// eslint-disable-next-line @typescript-eslint/ban-types
interface PreserveAliasName extends String {}

export interface KeyWithSalt {
  key: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type StringifiedKeyWithSalt = string & PreserveAliasName

export interface EncryptedObject {
  secretbox: Uint8Array
  nonce: Uint8Array
  salt: Uint8Array
}
/** Redux can't serialize Uint8Arrays so we base64 encode and stringify */
export type EncryptedString = string & PreserveAliasName
// Hover to check if inferred variable type is EncryptedString (not string)
export const testPreserveAliasName = '{}' as EncryptedString

export interface SetUnlockedRootStatePayload {
  persistedRootState: PersistedRootState
  stringifiedEncryptionKey: StringifiedKeyWithSalt
}

/* --- STATE --- */
export interface PersistState {
  hasPersistedProfiles: boolean
  isPersistenceUnsupported: boolean
  loading: boolean
  stringifiedEncryptionKey: undefined | 'skipped' | StringifiedKeyWithSalt
  enteredWrongPassword: boolean
}

export interface PersistedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network'> {}
export interface SyncedRootState extends Pick<RootState, 'theme' | 'wallet' | 'network' | 'persist'> {}
