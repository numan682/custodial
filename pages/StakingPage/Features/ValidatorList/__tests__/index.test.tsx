import { NodeInternal } from '@oasisprotocol/client/dist/client'
import { act, render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { stakingActions } from 'app/state/staking'
import { Validator } from 'app/state/staking/types'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { ValidatorList } from '..'
import { ThemeProvider } from '../../../../../../styles/theme/ThemeProvider'

jest.mock('@oasisprotocol/client/dist/client')

const activeValidator: Validator = {
  address: 'oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz',
  nodeAddress: 'oasis1qpwrtzadlddfuvdhjlta85268ju7dj0flsrfgg5x',
  current_rate: 0.1,
  rank: 0,
  status: 'active',
  escrow: 1000n.toString(),
  name: 'test-validator1',
  media: {
    email_address: 'test@test.com',
    tg_chat: 'telegram',
    twitter_acc: 'https://twitter.com/my_twitter',
    website_link: 'https://test.com',
  },
}
const inactiveValidator: Validator = {
  address: 'oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe',
  nodeAddress: 'oasis1qrdlqcv3tnv7qzuucnq0fncua52n66x7n5pm3n93',
  current_rate: 0.2,
  rank: 1,
  status: 'inactive',
  escrow: 1000n.toString(),
  name: 'test-validator2',
  media: activeValidator.media,
}
const unknownValidator: Validator = {
  address: 'oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7',
  nodeAddress: 'oasis1qq672rjh7mldhaj35mlf4w34m8jtl9vu2c8qspkz',
  current_rate: 0.2,
  rank: 2,
  status: 'unknown',
  escrow: 1000n.toString(),
  name: 'test-validator3',
  media: activeValidator.media,
}

const renderComponent = (store: any) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <ValidatorList />
      </ThemeProvider>
    </Provider>,
  )

describe('<ValidatorList  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore({
      network: {
        chainContext: '',
        epoch: 300,
        selectedNetwork: 'mainnet',
        ticker: 'ROSE',
        minimumStakingAmount: 100,
      },
    })

    jest.mocked(NodeInternal).mockReturnValue({
      async stakingAccount(query: any) {
        return {
          escrow: {
            commission_schedule: {
              bounds: [{ start: 0, rate_max: new Uint8Array([5]) }],
            },
          },
        }
      },
    } as NodeInternal)
  })

  it('empty should match snapshot', () => {
    const component = renderComponent(store)
    act(() => {
      store.dispatch(
        stakingActions.updateValidators({
          timestamp: new Date('2022').getTime(),
          network: 'mainnet',
          list: [],
        }),
      )
    })
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('list should match snapshot', () => {
    const component = renderComponent(store)
    act(() => {
      store.dispatch(
        stakingActions.updateValidators({
          timestamp: new Date('2022').getTime(),
          network: 'mainnet',
          list: [activeValidator, inactiveValidator, unknownValidator],
        }),
      )
    })
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should display validator details on click', async () => {
    renderComponent(store)
    act(() => {
      store.dispatch(
        stakingActions.updateValidators({
          timestamp: new Date('2022').getTime(),
          network: 'mainnet',
          list: [activeValidator],
        }),
      )
    })

    let row = screen.getByText(/test-validator/)
    expect(row).toBeVisible()
    await userEvent.click(row)

    const details = await screen.findByTestId('validator-item')
    row = screen.getAllByText(/test-validator/)[0]

    expect(details).toBeVisible()
    await userEvent.click(row)
    await waitFor(() => expect(details).not.toBeVisible())
  })

  it('should only display the details of a single validator', async () => {
    renderComponent(store)
    act(() => {
      store.dispatch(
        stakingActions.updateValidators({
          timestamp: new Date('2022').getTime(),
          network: 'mainnet',
          list: [activeValidator, inactiveValidator],
        }),
      )
    })

    let row = screen.getByText(/test-validator1/)
    expect(row).toBeVisible()
    await userEvent.click(row)

    let details = await screen.findByTestId('validator-item')
    expect(screen.getByTestId('validator-item-name').textContent).toEqual('test-validator1')
    expect(details).toBeVisible()

    row = screen.getByText(/test-validator2/)
    await userEvent.click(row)

    details = screen.getByTestId('validator-item')
    expect(await screen.findByTestId('validator-item-name')).toHaveTextContent('test-validator2')
    expect(details).toBeVisible()
  })
})
