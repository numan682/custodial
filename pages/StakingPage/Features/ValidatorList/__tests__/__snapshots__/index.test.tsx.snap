// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidatorList  /> empty should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #16213E;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c2 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c5 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: unset;
  background-color: unset;
}

.c3 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c4 {
  position: relative;
  width: 100%;
  display: table;
}

.c6 {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: unset;
  background-color: unset;
}

.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

<div
  class="c0"
  style="min-height: 100vh;"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      common.validators
    </h1>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5 rdt_Table"
          role="table"
        >
          <div
            class="c6"
          >
            <div
              style="padding: 24px;"
            >
              There are no records to display
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ValidatorList  /> list should match snapshot 1`] = `
.c12 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c12 g {
  fill: inherit;
  stroke: inherit;
}

.c12 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c12 *[stroke*='#'],.c12 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c12 *[fill-rule],
.c12 *[FILL-RULE],
.c12 *[fill*='#'],.c12 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c19 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #2ad5ab;
  stroke: #2ad5ab;
}

.c19 g {
  fill: inherit;
  stroke: inherit;
}

.c19 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c19 *[stroke*='#'],.c19 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c19 *[fill-rule],
.c19 *[FILL-RULE],
.c19 *[fill*='#'],.c19 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c21 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #d24c00;
  stroke: #d24c00;
}

.c21 g {
  fill: inherit;
  stroke: inherit;
}

.c21 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c21 *[stroke*='#'],.c21 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c21 *[fill-rule],
.c21 *[FILL-RULE],
.c21 *[fill*='#'],.c21 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c20 {
  font-size: 18px;
  line-height: 24px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #16213E;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c2 {
  margin-top: 0px;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c5 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: unset;
  background-color: unset;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  background-color: unset;
  min-height: 56px;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
  border-bottom-style: solid;
}

.c8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  font-size: 18px;
  font-weight: 500;
  color: unset;
  padding-left: 16px;
  padding-right: 16px;
}

.c17 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: 16px;
  padding-right: 16px;
  word-break: break-word;
}

.c9 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 34px;
  max-width: 34px;
}

.c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c13 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 110px;
  max-width: 110px;
}

.c18 {
  font-weight: 400;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: 100%;
  box-sizing: border-box;
  font-size: 16x;
  color: unset;
  background-color: unset;
  min-height: 48px;
}

.c16:not(:last-of-type) {
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
}

.c16:hover {
  color: unset;
  background-color: #88888833;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-bottom-color: unset;
  outline-style: solid;
  outline-width: 1px;
  outline-color: unset;
}

.c16:hover {
  cursor: pointer;
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c11:focus {
  outline: none;
  color: unset;
}

.c11:hover {
  color: unset;
}

.c11 span.__rdt_custom_sort_icon__ i,
.c11 span.__rdt_custom_sort_icon__ svg {
  opacity: 0;
  color: inherit;
  font-size: 18px !important;
  height: 18px !important;
  width: 18px !important;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-duration: 125ms;
  transition-duration: 125ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c11 span.__rdt_custom_sort_icon__.asc i,
.c11 span.__rdt_custom_sort_icon__.asc svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c11:hover {
  cursor: pointer;
}

.c11:hover span,
.c11:hover span.__rdt_custom_sort_icon__ * {
  opacity: 1;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c4 {
  position: relative;
  width: 100%;
  display: table;
}

.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-top: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media screen and (max-width:599px) {
  .c13 {
    display: none;
  }
}

@media screen and (max-width:599px) {
  .c14 {
    display: none;
  }
}

<div
  class="c0"
  style="min-height: 100vh;"
>
  <div
    class="c1"
  >
    <h1
      class="c2"
    >
      common.validators
    </h1>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5 rdt_Table"
          role="table"
        >
          <div
            class="c6 rdt_TableHead"
            role="rowgroup"
          >
            <div
              class="c7 rdt_TableHeadRow"
              role="row"
            >
              <div
                class="c8 c9 rdt_TableCol"
              />
              <div
                class="c8 c9 rdt_TableCol"
              />
              <div
                class="c8 c10 rdt_TableCol"
              >
                <div
                  class="c11 rdt_TableCol_Sortable"
                  id="column-name"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    validator.name
                  </div>
                  <span
                    class="asc __rdt_custom_sort_icon__"
                  >
                    <svg
                      aria-label="Down"
                      class="c12"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m2 8.35 10.173 9.823L21.997 8"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="c8 c13 rdt_TableCol"
              >
                <div
                  class="c11 rdt_TableCol_Sortable"
                  id="column-escrow"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    validator.escrow
                  </div>
                  <span
                    class="asc __rdt_custom_sort_icon__"
                  >
                    <svg
                      aria-label="Down"
                      class="c12"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m2 8.35 10.173 9.823L21.997 8"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="c8 c14 rdt_TableCol"
              >
                <div
                  class="c11 rdt_TableCol_Sortable"
                  id="column-fee"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    validator.fee
                  </div>
                  <span
                    class="asc __rdt_custom_sort_icon__"
                  >
                    <svg
                      aria-label="Down"
                      class="c12"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m2 8.35 10.173 9.823L21.997 8"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c15 rdt_TableBody"
            offset="250px"
            role="rowgroup"
          >
            <div
              class="c16 rdt_TableRow"
              id="row-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="row"
            >
              <div
                class="c17 c9 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-icon-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              />
              <div
                class="c17 c9 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-status-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                <svg
                  aria-label="Status is okay"
                  class="c19"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zM7 12l4 3 5-7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c17 c10 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-name-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                test-validator1
              </div>
              <div
                class="c17 c13 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-escrow-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                <span>
                  <span>
                    0
                  </span>
                  <span
                    class="c20"
                  >
                    <span
                      class="notranslate"
                      translate="no"
                    >
                       ROSE
                    </span>
                  </span>
                </span>
              </div>
              <div
                class="c17 c14 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-fee-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                10%
              </div>
            </div>
            <div
              class="c16 rdt_TableRow"
              id="row-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="row"
            >
              <div
                class="c17 c9 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-icon-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              />
              <div
                class="c17 c9 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-status-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                <svg
                  aria-label="Status is unknown"
                  class="c21"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2 3.99C2 2.892 2.898 2 3.99 2h16.02C21.108 2 22 2.898 22 3.99v16.02c0 1.099-.898 1.99-1.99 1.99H3.99A1.995 1.995 0 0 1 2 20.01V3.99zM12 15v-1c0-1 0-1.5 1-2s2-1 2-2.5c0-1-1-2.5-3-2.5s-3 1.264-3 3m3 6v2"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c17 c10 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-name-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                test-validator3
              </div>
              <div
                class="c17 c13 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-escrow-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                <span>
                  <span>
                    0
                  </span>
                  <span
                    class="c20"
                  >
                    <span
                      class="notranslate"
                      translate="no"
                    >
                       ROSE
                    </span>
                  </span>
                </span>
              </div>
              <div
                class="c17 c14 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-fee-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                20%
              </div>
            </div>
            <div
              class="c16 rdt_TableRow"
              id="row-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="row"
            >
              <div
                class="c17 c9 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-icon-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              />
              <div
                class="c17 c9 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-status-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                <svg
                  aria-label="Status is critical"
                  class="c21"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.703 2.703a.99.99 0 0 0-1.406 0l-8.594 8.594a.99.99 0 0 0 0 1.406l8.594 8.594a.99.99 0 0 0 1.406 0l8.594-8.594a.99.99 0 0 0 0-1.406l-8.594-8.594zM8.983 14.7 14.7 8.983m-5.717 0L14.7 14.7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c17 c10 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-name-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                test-validator2
              </div>
              <div
                class="c17 c13 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-escrow-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                <span>
                  <span>
                    0
                  </span>
                  <span
                    class="c20"
                  >
                    <span
                      class="notranslate"
                      translate="no"
                    >
                       ROSE
                    </span>
                  </span>
                </span>
              </div>
              <div
                class="c17 c14 c18 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-fee-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                20%
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
